snomed/com.b2international.snowowl.snomed.datastore/src/com/b2international/snowowl/snomed/datastore/request/SnomedComponentSearchRequest.java

```diff
@@ -43,21 +40,9 @@
 		
 	}
 	
-	protected final void addActiveMemberOfClause(ExpressionBuilder queryBuilder) {
-		if (containsKey(OptionKey.ACTIVE_MEMBER_OF)) {
-			final Collection<String> refSetIds = getCollection(OptionKey.ACTIVE_MEMBER_OF, String.class);
-			
-			final Expression referringRefSetExpression = SnomedComponentDocument.Expressions.referringRefSets(refSetIds);
-			final Expression referringMappingRefSetExpression = SnomedComponentDocument.Expressions.referringMappingRefSets(refSetIds);
-			
-			final Expression expression = Expressions
-					.builder()
-					.should(referringRefSetExpression)
-					.should(referringMappingRefSetExpression)
-					.build();
-				
-			queryBuilder.filter(expression);
-		}
+	protected final void addActiveMemberOfClause(BranchContext context, ExpressionBuilder queryBuilder) {
+		addEclFilter(context, queryBuilder, SnomedComponentSearchRequest.OptionKey.ACTIVE_MEMBER_OF, SnomedComponentDocument.Expressions::referringRefSets);
+		addEclFilter(context, queryBuilder, SnomedComponentSearchRequest.OptionKey.ACTIVE_MEMBER_OF, SnomedComponentDocument.Expressions::referringMappingRefSets);
```

Comment:
The two expressions created by the `addEclFilter` calls should be wrapped into a boolean expression with two should clauses.